clang-tools-extra/clang-doc/Representation.h

```diff
@@ -312,6 +317,16 @@ struct SymbolInfo : public Info {
 
   std::optional<Location> DefLoc;     // Location where this decl is defined.
   llvm::SmallVector<Location, 2> Loc; // Locations where this decl is declared.
+
+  bool operator<(const SymbolInfo &Other) const {
+    if (DefLoc && Other.DefLoc) {
+      return *DefLoc < *Other.DefLoc;
+    }
+    if (Loc.size() > 0 && Other.Loc.size() > 0) {
+      return Loc[0] < Other.Loc[0];
+    }
+    return extractName() < Other.extractName();
```

Comment:
Looking at the overall comparison, you could probably use `std::tuple` to make the logic simpler. The optional value may make that harder though, as I don't recall offhand how that's normally handled.

Reply:
https://en.cppreference.com/w/cpp/utility/tuple/operator_cmp